### ISSUE TITLE
File Chooser does not work in regolith-sway

### ISSUE THREAD

**Issue by gogsbread** [regolith-linux/regolith-desktop#1027]
- [x] I've [run a search on the existing bug reports](https://help.github.com/en/github/searching-for-information-on-github/searching-issues-and-pull-requests) to make sure I'm not filing a duplicate (you can use the search box at the top left corner)
- [x] I've read [the Regolith documentation](https://regolith-linux.org/docs/) and didn't find any relevant topic
- [ ] I tried [asking on Slack](https://regolith-linux.org/interact/) (especially for configuration related issues)

**Describe the bug**
File Chooser does not work in Regolith 3.2 Sway desktop (works in x11 desktop)

I understand that Regolith 3.2, with Ubuntu 24.04 support, is still in beta but you can use this for triaging. There is a similar issue(similar in error message) https://github.com/regolith-linux/regolith-desktop/issues/936  with a slightly different problem. I am opening this new bug since I am not sure what the status of the other one is given it has been open for more than 6 months

**To Reproduce**
- Use any application that opens a file. Say, Ctrl+o in Chrome
- File chooser will not launch
- `journalctl -f` shows error message `Backend call failed: No such interface “org.freedesktop.impl.portal.FileChooser” on object at path /org/freedesktop/portal/desktop`
- Unwrapping that error message, it looks like `/usr/libexec/xdg-desktop-portal-gnome` is running in some settings mode and hence it isn't handing the `FileChooser` dbus interface.
```
xdg-desktop-portal-gnome: Non-compatible display server, exposing settings only
```
- I then observed we have `/usr/share/xdg-desktop-portal/portals/regolith.portal` which claims to handle all interfaces but it doesn't even activate. There is also`xdg-desktop-portal-regolith` package that provides systemd service and `xdg-desktop-portal-regolith` binary but that implementation doesn't seem to be complete as gnome is the one that is activating.

**Expected behavior**
FIle chooser should launch

**Installation Details**
 - Regolith Install Type: [ISO/PPA] PPA
 - Regolith Version: [e.g. 1.4] 3.2
 - PPA url: [default: ppa:regolith-linux/release] testing-ubuntu-noble-amd64
 - Host OS (for PPA): [e.g. Ubuntu 20.04] Ubuntu 24.04

**Comment by kgilmer** [comment]
@SoumyaRanjanPatnaik is this expected behavior for sway session in 3.2?

**Comment by SoumyaRanjanPatnaik** [comment]
No, this is not the expected behaviour. @gogsbread can you check if you have `xdg-desktop-portal-gtk` installed ? Also, let me know what `echo $XDG_CURRENT_DESKTOP` prints.

**Comment by gogsbread** [comment]
- gtk is active
```
~λ scu is-active xdg-desktop-portal-gtk
active
```
- variables
```
~λ echo $XDG_CURRENT_DESKTOP; systemctl --user show-environment | grep XDG_CURRENT_DESKTOP
Regolith-Wayland:GNOME:sway
XDG_CURRENT_DESKTOP=Regolith-Wayland:GNOME:sway

```
- I couldn't find any `xdg-desktop-portal-wayland-config` packages. The package that is more aligned with sway is `xdg-desktop-portal-wlr`?

- As I observed earlier, `xdg-desktop-portal` is not activating gtk but it is activating gnome by virtue of `XDG_CURRENT_DESKTOP` having `GNOME`. I made xdg to activate `gtk`/`wlr` by changing `Environment` of `xdg-desktop-portal.service`. After this change,  I see the filechooser,  however, the chooser don't seem to respect gnome theming.
```
[Service]
Environment=XDG_CURRENT_DESKTOP=gtk
```
```
Jun 29 22:11:14 systemd[1]: systemd-hostnamed.service: Deactivated successfully.
Jun 29 22:11:15 dbus-daemon[17575]: [session uid=1000 pid=17575] Activating via systemd: service name='org.freedesktop.impl.portal.desktop.gnome' unit='xdg-desktop-portal-gnome.service' requested by ':1.89' (uid=1000 pid=18612 comm="/usr/libexec/xdg-desktop-portal" label="unconfined")
Jun 29 22:11:15 systemd[17527]: Starting xdg-desktop-portal-gnome.service - Portal service (GNOME implementation)...
Jun 29 22:11:15 dbus-daemon[17575]: [session uid=1000 pid=17575] Successfully activated service 'org.freedesktop.impl.portal.desktop.gnome'
Jun 29 22:11:15 systemd[17527]: Started xdg-desktop-portal-gnome.service - Portal service (GNOME implementation).
```

**Comment by develNor** [comment]
I had a similar [issue](https://github.com/regolith-linux/regolith-desktop/issues/1022) that the filebrowser did not launch in Regolith 3.2 under Ubuntu 24.04.

In my case it was solved by using `unstable` stage instead of of the `testing` stage, so this might be related to something that was recently fixed. For Debugging here is more information about my used versions:

**Version with Problem**
- ppa: `testing-ubuntu-noble-amd64 noble main`
- installed around the 17.06
- => Beta 1

**Version without Problem**
- ppa: `unstable-ubuntu-noble-amd64 noble main`
- installed on the 21.06

**Comment by SoumyaRanjanPatnaik** [comment]
> - gtk is active
> ```
> ~λ scu is-active xdg-desktop-portal-gtk
> active
> ```
> - variables
> ```
> ~λ echo $XDG_CURRENT_DESKTOP; systemctl --user show-environment | grep XDG_CURRENT_DESKTOP
> Regolith-Wayland:GNOME:sway
> XDG_CURRENT_DESKTOP=Regolith-Wayland:GNOME:sway
> 
> ```
> - I couldn't find any `xdg-desktop-portal-wayland-config` packages. The package that is more aligned with sway is `xdg-desktop-portal-wlr`?
> 
> - As I observed earlier, `xdg-desktop-portal` is not activating gtk but it is activating gnome by virtue of `XDG_CURRENT_DESKTOP` having `GNOME`. I made xdg to activate `gtk`/`wlr` by changing `Environment` of `xdg-desktop-portal.service`. After this change,  I see the filechooser,  however, the chooser don't seem to respect gnome theming.
> ```
> [Service]
> Environment=XDG_CURRENT_DESKTOP=gtk
> ```
> ```
> Jun 29 22:11:14 systemd[1]: systemd-hostnamed.service: Deactivated successfully.
> Jun 29 22:11:15 dbus-daemon[17575]: [session uid=1000 pid=17575] Activating via systemd: service name='org.freedesktop.impl.portal.desktop.gnome' unit='xdg-desktop-portal-gnome.service' requested by ':1.89' (uid=1000 pid=18612 comm="/usr/libexec/xdg-desktop-portal" label="unconfined")
> Jun 29 22:11:15 systemd[17527]: Starting xdg-desktop-portal-gnome.service - Portal service (GNOME implementation)...
> Jun 29 22:11:15 dbus-daemon[17575]: [session uid=1000 pid=17575] Successfully activated service 'org.freedesktop.impl.portal.desktop.gnome'
> Jun 29 22:11:15 systemd[17527]: Started xdg-desktop-portal-gnome.service - Portal service (GNOME implementation).
> ```

Hey sorry. The package name is `xdg-desktop-portal-regolith-wayland-config`

**Comment by gogsbread** [comment]
> In my case it was solved by using the unstable stage instead of the testing stage, so this might be related to something that was recently fixed. 

Thanks. This does help with the File Chooser issue. However, this update(or probably just regolith-sway) seems to introduce a theming issue. My dark theme switches back to light theme and not amount of  `regolith-look set <>; regolith-look refresh` is fixing this. I also feel this is beyond theming and that gnome settings that you apply in gnome control center is also not working; for ex, my two finger right-click is gone.
